### PR TITLE
[FIX] sale: prevent modification of discount on locked sale orders

### PR DESCRIPTION
**Pull request by lef-adhoc**
Description of the issue/feature this PR addresses:

When a sale order is opened in multiple tabs and confirmed in one (locking it), other tabs still allow modifications like changes to the `discount` field. This leads to data inconsistencies, as locked orders should prevent further edits.

Current behavior before PR:

The `discount` field can still be modified in other tabs after a sale order is locked, causing inconsistencies.

Desired behavior after PR is merged:

The `discount` field is now protected on locked sale orders, preventing any modifications and ensuring data consistency.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Feyensv** [comment]
It makes sense, but I don't believe it's safe to change that in stable, could you please retarget the PR to only change that in master?
That way, we'll be sure to avoid breaking existing code.

**Comment by lef-adhoc** [comment]
@Feyensv Thank you for the feedback. I've retargeted the PR to master as requested

**Comment by Feyensv** [comment]
Will have to wait until your CLA PR has been merged and forward-ported to the master branch ;)

**Review by Feyensv — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**
